### PR TITLE
Jacoco documentation: fix a small issue

### PR DESCRIPTION
**Pull request by tasomaniac**
### Context

While integrating the Jacoco plugin into our repo, I realized a small build performance issue with the suggestion provided in the docs.

When `test` task is always finalized with `jacocoTestReport`, it causes jacoco report to be generated when not needed. Example: Recent versions of IntelliJ IDEA delegates to Gradle to run a single test. This is done during development workflow and supposed to be very fast. In this case, we don't need to generate jacoco report every time a single test is run.

### Solution

Remove the suggestion to make tests being finalized by jacoco task. Making `jacocoTestReport` depend on the test task should be enough. 

Also change the example usage to use Task configuration avoidance APIs.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] *N/A* Provide** integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] *N/A* Provide** unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [x] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [x] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
I'd rather document the performance impact over removing the suggestion. Can you instead update the documentation with a better solution?

**Comment by tasomaniac** [comment]
I cannot see how it can be improved since it is just simply unnecessary. With my suggestion, it is simpler. If you want jaco report, you simply run jaco tasks and they automatically depend on tests. All works fine. 

If you have any suggestions, I'm open to change it.

**Comment by JLLeitschuh** [comment]
🤔 Perhaps suggest that they attach the jacoco tasks to `check`? Maybe?

**Comment by JLLeitschuh** [comment]
@big-guy your thoughts here?

**Comment by tasomaniac** [comment]
That's a good idea. I will change it.

**Comment by JLLeitschuh** [comment]
Thanks @tasomaniac!

**Comment by JLLeitschuh** [comment]
@bot-gradle help

**Comment by JLLeitschuh** [comment]
@bot-gradle test SanityCheck plz

**Comment by JLLeitschuh** [comment]
@bot-gradle test QuickFeedback

**Comment by tasomaniac** [comment]
Let's keep this open. It is small but very useful documentation change

**Comment by tasomaniac** [comment]
Any updates on this? Can someone trigger the CI again? There is not much reason why this would fail in the CI?

File: subprojects/docs/src/snippets/testing/jacoco-quickstart/groovy/build.gradle [hunk 1]
```diff
@@ -33,10 +33,10 @@ test {
 // end::testtask-configuration[]
 
 // tag::testtask-dependency[]
-test {
-    finalizedBy jacocoTestReport // report is always generated after tests run
+tasks.named("check").configure {
```
Comment (ljacomet) [inline subprojects/docs/src/snippets/testing/jacoco-quickstart/groovy/build.gradle:36]: These changes make no sense in isolation. The text before the inclusion of that snippet should be adapted as well then.

Another option is to make it clearer here that what is proposed is _optional_

**Review by ljacomet — CHANGES_REQUESTED** [review]
Please make snippet changes in the context of where this is included.
See https://docs.gradle.org/current/userguide/jacoco_plugin.html#sec:jacoco_getting_started and the text before example 2

**Comment by donat** [comment]
I'm closing this PR as there hasn't been any progress for quite a while now. @tasomaniac let me know if you are still interested in finishing this work.

**Pull request closed without merging**